SPECS/rpm/CVE-2021-20266.patch

```diff
@@ -0,0 +1,59 @@
+From 8ec31f3cf4071b4457d39adf10c65fb0a00657e5 Mon Sep 17 00:00:00 2001
+From: "Demi M. Obenour" <demiobenour@gmail.com>
+Date: Fri, 1 Jan 2021 12:56:13 -0500
+Subject: [PATCH] Re-check header in hdrblobInit()
+
+Functions in the public RPM API use hdrblobInit() to import an RPM
+header from memory, but that skips critical checks.  Fix this by not
+skipping these checks.
+---
+ lib/header.c | 30 ++++++++++++++++++++++++++++++
+ 1 file changed, 30 insertions(+)
+
+diff --git a/lib/header.c b/lib/header.c
+index 0c450bea2b..cbf944515b 100644
+--- a/lib/header.c
++++ b/lib/header.c
+@@ -1968,12 +1968,42 @@ rpmRC hdrblobInit(const void *uh, size_t uc,
+ 		rpmTagVal regionTag, int exact_size,
+ 		struct hdrblob_s *blob, char **emsg)
+ {
++    size_t nb;
```

Comment:
I'm not sure if this is a full patch. When I read through the [bug discussion](https://bugzilla.redhat.com/show_bug.cgi?id=1927741), this is the [first version of the patch](https://github.com/rpm-software-management/rpm/pull/1500/files) and later on Todd Cullum seems to be sending an [updated patch](https://github.com/rpm-software-management/rpm/commit/8f4b3c3cab8922a2022b9e47c71f1ecf906077ef). Can you double-check?

Reply:
You're right, I've double-checked and missed that it modifies some functions later on. I've merged the two patches.

Reply:
That's not exactly what I had in mind. Notice that the first version I've mentioned above has been rejected (the author has closed the PR and it was never merged). Considering that, I **believe** the updated patch is the correct one, but that's what I'd like you to double-check by closely reading the bug discussion.

Reply:
+1 here. From reading through the bug discussion, Todd Cullum's link to the [upstream patch (commit id 8f4b3c)](https://github.com/rpm-software-management/rpm/commit/8f4b3c3cab8922a2022b9e47c71f1ecf906077ef) is the only one we should use.

Reply:
Thanks. I looked at the patch and it did not work with our release - few of the lines seems to have changed in the meantime. I inspected the modifications and modified the patch accordingly. The modified lines were either already updated or modified in a fashion that did not affect the patch.